pkg/cgroups/cgroups_linux.go

```diff
@@ -843,12 +843,14 @@ func UserOwnsCurrentSystemdCgroup() (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		s := st.Sys()
+
+		//nolint:errcheck
+		s := st.Sys().(*syscall.Stat_t)
```

Comment:
> For an expression x of [interface type](https://go.dev/ref/spec#Interface_types), but not a [type parameter](https://go.dev/ref/spec#Type_parameter_declarations), and a type T, the primary expression
> 
> x.(T)
> **asserts that x is not nil** and that the value stored in x is of type T.

i.e. the `== nil` check must happen before the `.(*…)` cast, if it is expected to ever hit.

Reply:
yeah I see, I wasn't sure what the linter wanted me to do there exactly. I assumed it doesn't like the cast but you are right Sys() might be nil.

Not sure if we want to do that check then if we think it is impossible, I rather have a panic with the stack trace on how we got here.

Reply:
ACK.